src/main/java/org/opensearch/security/action/apitokens/ApiTokenRepository.java

```diff
@@ -17,19 +17,36 @@
 import org.opensearch.client.Client;
 import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.security.authtoken.jwt.ExpiringBearerAuthToken;
+import org.opensearch.security.identity.SecurityTokenManager;
 
 public class ApiTokenRepository {
     private final ApiTokenIndexHandler apiTokenIndexHandler;
+    private final SecurityTokenManager securityTokenManager;
 
-    public ApiTokenRepository(Client client, ClusterService clusterService) {
+    public ApiTokenRepository(Client client, ClusterService clusterService, SecurityTokenManager tokenManager) {
         apiTokenIndexHandler = new ApiTokenIndexHandler(client, clusterService);
+        securityTokenManager = tokenManager;
     }
 
-    public String createApiToken(String name, List<String> clusterPermissions, List<ApiToken.IndexPermission> indexPermissions) {
+    public ApiTokenRepository(ApiTokenIndexHandler apiTokenIndexHandler1, SecurityTokenManager tokenManager) {
```

Comment:
This is a constructor that I created just for testing to inject a mocked apiTokenIndexHandler. Otherwise I had to do some hacky things with setting it via reflection. I'm not sure about the standards around this - is it bad to create a constructor that is only used in testing?

Reply:
Yes, general practice is to not create public constructors. Ideally you should create a static method, say `forTest()` that returns a test constructor.

Reply:
Fixed in: [08e0890](https://github.com/opensearch-project/security/pull/4967/commits/08e08900b914f7df626692108b8099a745343bee) according to suggestions.